src/entities/bond.ts

```diff
@@ -12,6 +12,8 @@ export interface TrancheData {
     index: string;
     ratio: string;
     totalCollateral: BigNumberish;
+    totalCollateralAtMaturity: BigNumberish;
```

Comment:
Why do we need these on top of the bond data ones -- I think these can be derived from the bond-level fields

Reply:
also shouldn't they be nullable?

Reply:
I think the most sensible minimal data approach would be omitting bond.totalCollateralAtMaturity from the underlying data and having the getter sum the tranche values.
Trying to calculate the tranche ones based on the bond total and cdrs might be less accurate, haven't fully thought the rounding math through.

I kept both in underlying data for symmetry's sake and reduced work (don't need to sum every time the getter is called) but happy to remove bond,totalCollateralAtMaturity from the underlying data.